spec/forms/candidate_interface/contact_details_form_spec.rb

```diff
@@ -61,20 +61,46 @@
       expect(application_form.international_address).to be_nil
     end
 
-    it 'updates the provided ApplicationForm with the international address field if valid' do
-      form_data = {
-        international_address: '123 Chandni Chowk, Old Delhi',
-      }
-      application_form = build(:application_form)
-      contact_details = CandidateInterface::ContactDetailsForm.new(form_data)
+    # TODO: remove this context when removing `international_addresses` feature flag
+    context 'without international_addresses feature flag' do
+      before { FeatureFlag.deactivate(:international_addresses) }
+
+      it 'updates the provided ApplicationForm with the international address field if valid' do
+        form_data = {
+          international_address: '123 Chandni Chowk, Old Delhi',
+        }
+        application_form = build(:application_form)
+        contact_details = CandidateInterface::ContactDetailsForm.new(form_data)
+
+        expect(contact_details.save_address(application_form)).to eq(true)
+        expect(application_form).to have_attributes(form_data)
+        expect(application_form.address_line1).to be_nil
+        expect(application_form.address_line2).to be_nil
+        expect(application_form.address_line3).to be_nil
+        expect(application_form.address_line4).to be_nil
+        expect(application_form.postcode).to be_nil
+      end
+    end
 
-      expect(contact_details.save_address(application_form)).to eq(true)
-      expect(application_form).to have_attributes(form_data)
-      expect(application_form.address_line1).to be_nil
-      expect(application_form.address_line2).to be_nil
-      expect(application_form.address_line3).to be_nil
-      expect(application_form.address_line4).to be_nil
-      expect(application_form.postcode).to be_nil
+    context 'with international_addresses feature flag' do
+      before { FeatureFlag.activate(:international_addresses) }
+
+      it 'updates the provided ApplicationForm with the international address field if valid' do
+        form_data = {
+          address_line1: '123 Chandni Chowk',
+          address_line3: 'Old Delhi',
+          postcode: '110006',
+        }
+        application_form = build(:application_form)
+        contact_details = CandidateInterface::ContactDetailsForm.new(form_data)
```

Comment:
Should we be using the `build_from_application` method here instead? And if not, are you able to explain why please?

Reply:
:+1: It's a bit simpler to just pass the form_data into set it up but it's a bit different to the way this would be used in real life so I've changed it to user `build_from_application` to initialise the form from a model, then user `assign_attributes` to input the form_data.

Reply:
Switched back to original following call. (`build_from_application` isn't used in `update` actions so the original was closer to real life usage).